components/job-orchestration/job_orchestration/retention/streams_handler.py

```
@@ -0,0 +1,93 @@
+            object_ids_to_delete.append(stream.get(MONGODB_ID_KEY))
+
+        targets_buffer.persists_new_targets()
+
+        # TODO: here we don't use retention_filter again to avoid race condition between
+        # targets to delete and timestamp of file.
+        # This could create another race condition that
+        # 1. daemon mark the entry as stale
+        # 2. somehow webui access the entry and updates its last access time.
+        # 3. daemon removes the file while log viewer tries to load it, race condition. boom
+        # There are some strategy we can take to prevent this.
+        # 1. Simply sleep for around ~10 seconds before deleting the files so most probably log viewer
```

Pylint (3.3.7)

[convention] Line too long (102/100)

(C0301)